### PR TITLE
Allow for strings

### LINKED ISSUE #11: Add helper for creating resources from strings
By forcing streams to be created with existing resources, (see https://github.com/http-interop/http-factory/pull/2#issuecomment-240454239), it makes the factory more difficult to use.

While I approve of the reasoning that @weierophinney makes, we should also seek to keep the barrier to entry low enough that developers do not have to learn how to create a temporary resource to use the stream factory. I think it would be worth defining a function that is included with this package to make it easier:

``` php
function get_temporary_resource(string $content) : resource
```

Usage would make calls to factories more clear:

``` php
$stream = $streamFactory->createSream(get_temporary_resource($content));
```

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #11

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
👍

File: src/StreamFactoryInterface.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
      *
      * The resource MUST be readable and SHOULD be seekable. It MAY be writable.
```
Comment (shadowhand) [inline src/StreamFactoryInterface.php]: This comment needs to be changed.

**Comment by Nyholm** [comment]
If you have any suggestions about the comment, please share. 
I saw that the previous comment was: 

> The stream will be writable and seekable.

I've made an update now.

File: src/StreamFactoryInterface.php [hunk 2]
```diff
@@ -9,9 +9,9 @@
     /**
      * Create a new stream from a resource.
```
Comment (shadowhand) [inline src/StreamFactoryInterface.php]: This comment is also incorrect if a string is allowed. Original implementation: https://github.com/http-interop/http-factory/blob/e4eee1d5c140e699005999671340d6d360f26dff/src/StreamFactoryInterface.php

File: src/StreamFactoryInterface.php [hunk 3]
```diff
@@ -7,11 +7,11 @@
 interface StreamFactoryInterface
 {
     /**
-     * Create a new stream from a resource.
+     * Create a new stream from a resource or a string.
      *
-     * The resource MUST be readable and SHOULD be seekable. It MAY be writable.
+     * If argument is resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
```
Comment (shadowhand) [inline src/StreamFactoryInterface.php]: I would also like to see a note about strings:

> If the argument is a string, a temporary resource will be created that is writable and seekable.

File: src/StreamFactoryInterface.php [hunk 4]
```diff
@@ -9,7 +9,8 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If argument is resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
+     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
+     * If the argument is a string, a temporary resource will be created that is writable and seekable.
```
Comment (shadowhand) [inline src/StreamFactoryInterface.php:13]: Perfect, thanks!

**Comment by shadowhand** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/http-interop/http-factory/pull-request/12/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=http-interop/http-factory)

**Comment by Nyholm** [comment]
Thank you for merging.

File: src/StreamFactoryInterface.php [hunk 5]
```diff
@@ -7,11 +7,12 @@
...
     /**
-     * Create a new stream from a resource.
+     * Create a new stream from a resource or a string.
      *
-     * The resource MUST be readable and SHOULD be seekable. It MAY be writable.
+     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
+     * If the argument is a string, a temporary resource will be created that is writable and seekable.
      *
-     * @param resource $resource
+     * @param string|resource $resource
```
Comment (harikt) [inline src/StreamFactoryInterface.php:15]: I would say what sort of string can be passed should be defined.

Comment (shadowhand) [inline src/StreamFactoryInterface.php:15]: Good point. @Nyholm would you be willing to submit another PR that makes it clear the string cannot be a filename?

Comment (Nyholm) [inline src/StreamFactoryInterface.php:15]: Of course.
